lib/source/encoding.pm

```diff
@@ -0,0 +1,72 @@
+package source::encoding;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.01';
+
+our $ascii_hint_bits = 0x00000010;
+
+sub import {
+    unimport();     # Get rid of any 'use utf8'
+    my (undef, $arg) = @_;
+    if (lc $arg eq 'utf8') {
+        require utf8;
+        utf8->import;
+        return;
+    }
+    elsif (lc $arg eq 'ascii') {
+        $^H |= $ascii_hint_bits;
+        return;
+    }
+
+    die "Bad argument for source::encoding: '$arg'";
+}
+
+sub unimport {
+    $^H &= ~$ascii_hint_bits;
+    utf8->unimport;
+}
+
+1;
+__END__
+
+=head1 NAME
+
+source::encoding -- Declare Perl script's encoding
```

Comment:
Back in the day, we used to get annoyed when people referred to "Perl *scripts*" rather than "Perl *programs*."  I recommend using *program* rather than *script* unless there's a reason to do otherwise in a specific context.

Reply:
Is this only in a script? Does it also work in modules?

Reply:
I think "Perl file" would be better descriptive indeed, as this is neither restricted to script files nor applies to the whole program.

Reply:
Maybe go with something cute / monastic, like "Tome"?

Reply:
I used `source code`, to see how you like that

Reply:
It has a nice symmetry to it

Reply:
Works for me. Perhaps the description could mention explicitly that the pragma has a lexical effect on the current scope, so there is no ambiguity and no need to mention files.